lib/node/node.js

```diff
@@ -278,6 +292,21 @@ Node.prototype.getInputNodes = function() {
     return this.inputNodes;
 };
 
+Node.prototype.getAllInputNodes = function(filterFn) {
+    filterFn = filterFn || function() { return true; };
+
+    function getAllInputNodes(node, nodeList) {
+        node.getInputNodes().forEach(function(inputNode) {
+            nodeList.push(inputNode);
+            getAllInputNodes(inputNode, nodeList);
+        });
+
+        return nodeList;
+    }
+
+    return getAllInputNodes(this, []).filter(filterFn);
```

Comment:
Hum didn't think about having the same input for two different nodes, as you put in there so looks like we have to return a Set of nodes